infra/build/functions/update_build_status.py

```diff
@@ -233,19 +243,58 @@ def upload_log(build_id):
 # pylint: disable=no-member
 def update_status(event, context):
   """Entry point for cloud function to update build statuses and badges."""
-  del event, context  #unused
+  del context
+
+  if 'data' in event:
+    status_type = base64.b64decode(event['data']).decode()
+  else:
+    raise RuntimeError('No data')
+
+  if status_type == 'badges':
+    update_badges()
+    return
+
+  if status_type == 'fuzzing':
+    tag = build_project.FUZZING_BUILD_TAG
+    status_filename = FUZZING_STATUS_FILENAME
+  elif status_type == 'coverage':
+    tag = build_and_run_coverage.COVERAGE_BUILD_TAG
+    status_filename = COVERAGE_STATUS_FILENAME
+  else:
+    raise RuntimeError('Invalid build status type ' + status_type)
```

Comment:
nit: I'd personally prefer log / error messages in a format like: `'Invalid build status type:' + status_type` (a colon to separate string literal and the actual value, same for other prints / exceptions), but that's just a preference I guess and definitely isn't worth another CL to address